src/lib/dev-environment/dev-environment-cli.js

```diff
@@ -474,7 +498,7 @@ export async function promptForComponent( component: string, allowLocal: boolean
 	} );
 
 	let initialMode = 'image';
-	if ( 'appCode' === component ) {
+	if ( 'appCode' === component && isStdinTTY ) {
```

Comment:
`local` is dangerous when TTY is not available, as it is impossible to input the directory.